File: spring/core/src/main/java/stock/global/core/entities/Ticker.java
package stock.global.core.entities;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tickers")
public class Ticker{
    
    @Id
    @Column(name = "symbol", length = 12, nullable = false)
    private String symbol;

    @Column(name = "stock_type", length = 20)
    private String stockType;

    @Column(name = "name", length = 100)
    private String name;

    @Column(name = "ko_name", length = 100)
    private String koName;

    @Column(name = "nation_type", length = 12)
    private String nationType;

    @Column(name = "nation_code", length = 12)
    private String nationCode;

    @Column(name = "start_time", length = 20)
    private String startTime;

    @Column(name = "end_time", length = 20)
    private String endTime;

    @Column(name = "market_type", length = 20)
    private String marketType;

    @Column(name = "reuters_code", length = 20)
    private String reutersCode;

    @Column(name = "market_status", length = 10)
    private String marketStatus;

    @Column(name = "market_value", precision = 13, scale = 2)
    private BigDecimal marketValue;

    @Column(name = "stop_code", length = 20)
    private String stopCode;

    @Column(name = "stop_text", length = 100)
    private String stopText;

    @Column(name = "stop_name", length = 100)
    private String stopName;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @OneToMany(
        targetEntity = DayCandle.class,
        fetch = FetchType.LAZY ,
        mappedBy="ticker"
    )
    private List<DayCandle> dayCandles;
}

File: go/pkg/models/ticker.go
package models

import (
	"time"

	"github.com/jjh930301/stock_global/pkg/constant"
	"github.com/shopspring/decimal"
)

type Ticker struct {
	Symbol       string          `gorm:"column:symbol;type:varchar(5);not null;primary_key;comment:ticker symbol" json:"symbol,omitempty"`
	StockType    string          `gorm:"column:stock_type;type:varchar(20);default null;comment:check stock type" json:"stockType"`
	Name         string          `gorm:"column:name;type:varchar(100);default null;comment:name" json:"name"`
	KoName       string          `gorm:"column:ko_name;type:varchar(100);default null;comment:korea name" json:"koName"`
	NationType   string          `gorm:"column:nation_type;type:varchar(12)" json:"nationType"`
	NationCode   string          `gorm:"column:nation_code;type:varchar(12)" json:"nationCode"`              // StockExchangeInfo.NationCode
	StartTime    string          `gorm:"column:start_time;type:varchar(20);default null" json:"startTime"`   // StockExchangeInfo.StartTime
	EndTime      string          `gorm:"column:end_time;type:varchar(20);default null" json:"endTime"`       // StockExchangeInfo.EndTime
	MarKetType   string          `gorm:"column:market_type;type:varchar(20);default null" json:"marketType"` // StockExchangeInfo.Name
	ReutersCode  string          `gorm:"column:reuters_code;type:varchar(20);default null" json:"reuterCode"`
	MarketStatus string          `gorm:"column:market_status;type:varchar(10);default null;comment:market status CLOSE | ?" json:"marketStatus"`
	MarketValue  decimal.Decimal `gorm:"column:market_value;type:decimal(13,2);default 0;comment:시가총액" json:"marketValue"`
	StopCode     string          `gorm:"column:stop_code;type:varchar(20);default null;" json:"stopCode"`
	StopText     string          `gorm:"column:stop_text;type:varchar(100);default null;" json:"stopText"`
	StopName     string          `gorm:"column:stop_name;type:varchar(100);default null;" json:"stopName"`
	CreatedAt    time.Time       `gorm:"column:created_at;type:datetime;autoCreateTime;" json:"created_at" time_format:"unix"`
	DayCandles   []DayCandle     `gorm:"foreignKey:Symbol;references:Symbol" json:"dayCandles,omitempty"`
}

func (Ticker) TableName() string {
	return constant.Ticker
}
